codex/node.nim

```diff
@@ -543,6 +544,47 @@ proc onStore(
 
   return success()
 
+proc onProve(
+  self: CodexNodeRef,
+  slot: Slot,
+  challenge: ProofChallenge): Future[?!seq[byte]] {.async.} =
+  ## Generats a proof for a given slot and challenge
+  ##
+
+  let
+    cidStr = slot.request.content.cid
+    slotIdx = slot.slotIndex.truncate(Natural)
+
+  logScope:
+    cid = cidStr
+    slot = slotIdx
+    challenge = challenge
+
+  trace "Received proof challenge"
+
+  without cid =? Cid.init(cidStr).mapFailure, err:
+    error "Unable to parse Cid", cid, err = err.msg
+    return failure(err)
+
+  without manifest =? await self.fetchManifest(cid), err:
+    error "Unable to fetch manifest for cid", err = err.msg
+    return failure(err)
+
+  without builder =? SlotsBuilder.new(self.blockStore, manifest), err:
+    error "Unable to create slots builder", err = err.msg
+    return failure(err)
+
+  without sampler =? DataSampler.new(slotIdx, self.blockStore, builder), err:
+    error "Unable to create data sampler", err = err.msg
+    return failure(err)
+
+  without proofInput =? await sampler.getProofInput(challenge, nSamples = 3), err:
+    error "Unable to get proof input for slot", err = err.msg
+    return failure(err)
+
+  # Todo: send proofInput to circuit. Get proof. (Profit, repeat.)
+  success(@[42'u8])
```

Comment:
Yes!! ;)

Reply:
So here's where we'd want to call into the Rust Arkcom?

Reply:
yep